taichi/program/async_engine.cpp

```diff
@@ -216,28 +236,23 @@ void AsyncEngine::launch(Kernel *kernel, Context &context) {
   TI_ASSERT(block);
 
   auto &offloads = block->statements;
-  auto &kmeta = kernel_metas_[kernel];
-  const bool kmeta_inited = kmeta.initialized;
   for (std::size_t i = 0; i < offloads.size(); i++) {
     auto *offload = offloads[i]->as<OffloadedStmt>();
-    uint64 h;
-    OffloadedStmt *offl_template = nullptr;
-    if (kmeta_inited) {
-      auto &oc = kmeta.offloaded_cached[i];
-      h = oc.get_hash();
-      offl_template = oc.get_template();
+    uint64 h = ir_bank_.get_hash(offload);
```

Comment:
Note that previously, we compute the hash on the *cloned* IR, not the original one in the kernel. I'm not sure if this change could result in any problem, though..

Reply:
Good point. I just realized that the `stmt->id`s may be different -- the original ones in the kernel can be
```
kernel {
  $0 = offload {
    $1 ~ $9
  }
  $10 = offload {
    $11 ~ $20
  }
}
```
but the cloned ones will have `stmt->id` relabeled. I think we should reuse `KernelMeta` and compute the hash on the cloned IR...